### ISSUE TITLE
ECONNREFUSED 127.0.0.1:<PORT> on Windows 10

### ISSUE THREAD

**Issue by GavinRay97** [graalvm/vscode-extensions#10]
Below I have included as much information as possible.
This issue does not happen in WSL or in a Devcontainer with Ubuntu.

(Thank you so much to the Graal and Apache Netbeans folks for such an awesome platform + extension by the way)

```ps1
[Error - 10:37:27 AM] Starting client failed
Error: connect ECONNREFUSED 127.0.0.1:18189
	at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)
```
```ps1
Initiating server activation
Request to kill LSP server.
Cannot kill: current process is None
Launching Apache NetBeans Language Server with C:\GraalVM\graalvm-ce-java16-21.1.0-dev
LSP server launching: 14660
-------------------------------------------------------------------------------
>Log Session: Sunday, June 13, 2021 at 10:37:28 AM Eastern Daylight Time
>System Info: 
  Product Version         = vsc 12.4-21726744165c946ba6619bff89e98d5863f26e22
  Operating System        = Windows 10 version 10.0 running on amd64
  Java; VM; Vendor        = 16; OpenJDK 64-Bit Server VM 16+36-jvmci-21.1-b03; GraalVM Community
  Runtime                 = OpenJDK Runtime Environment 16+36-jvmci-21.1-b03
  Java Home               = C:\GraalVM\graalvm-ce-java16-21.1.0-dev
```
```json
{
	  "java.home": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev",
	  "netbeans.jdkhome": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev",
	  "graalvm.home": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev",
	  "graalvm.installations": ["C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev"],
	  "terminal.integrated.env.windows": {
	    "GRAALVM_HOME": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev",
	    "JAVA_HOME": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev",
	    "PATH": "C:\\GraalVM\\graalvm-ce-java16-21.1.0-dev\\bin;<TRUNCATED>"
	  }
}
```
![image](https://user-images.githubusercontent.com/26604994/122311009-470d1280-cedf-11eb-8b98-f67457e3bcf2.png)

```ps1
C:\Users\rayga>gu list
Illegal format in tzmappings file: illegal null character found at line 1, offset 47.
ComponentId              Version             Component name                Stability                     Origin
-------------------------------------------------------------------------------------------------------------------
graalvm                  21.1.0-dev          GraalVM Core                  -
js                       21.1.0-dev          Graal.js                      Early adopter (suppo
nfi                      21.1.0-dev          SVM Truffle NFI Support       Early adopter (suppo
```

![image](https://user-images.githubusercontent.com/26604994/122311040-55f3c500-cedf-11eb-9c71-5995feba7a51.png)
![image](https://user-images.githubusercontent.com/26604994/122311053-5a1fe280-cedf-11eb-83a0-19aa5f1a0d14.png)
![image](https://user-images.githubusercontent.com/26604994/122311060-5d1ad300-cedf-11eb-952d-a8175d60ddc0.png)
![image](https://user-images.githubusercontent.com/26604994/122311062-60ae5a00-cedf-11eb-814b-1335e8484866.png)

-----------------

Example VS Code Devcontainer configuration which works fine:

```dockerfile
FROM ubuntu:21.10
ENV DEBIAN_FRONTEND=noninteractive
​
# Gradle
ENV GRADLE_VERSION 7.1
ENV GRADLE_HOME /opt/gradle
VOLUME /home/gradle/.gradle
​
# GraalVM
ENV JAVA_HOME=/graalvm
ENV PATH=$JAVA_HOME/bin:$PATH
​
ARG JAVA_VERSION=16
ARG TARGET_ARCH=linux-amd64
​
# Grabbing a nightly release, 21.2.0-dev-20210615_0052
ARG GRAALVM_VERSION=21.2.0-dev
ARG GRAALVM_GITHUB_RELEASE_TAG=20210615_0052
​
# /graalvm-ce-java16-21.2.0-dev
ARG GRAALVM_HOME=/graalvm-ce-java${JAVA_VERSION}-${GRAALVM_VERSION}
# graalvm-ce-java16-linux-amd64-dev.tar.gz
ARG GRAALVM_DOWNLOAD_TARGET=graalvm-ce-java${JAVA_VERSION}-${TARGET_ARCH}-dev.tar.gz
ARG GRAALVM_DEV_DOWNLOADS_URL=https://github.com/graalvm/graalvm-ce-dev-builds/releases/download
​
RUN  set -o errexit -o nounset -o xtrace \
  && apt-get update \
  && apt-get install --no-install-recommends --yes   \
  unzip ca-certificates wget \
  # GraalVM
  # Example: https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/21.2.0-dev-20210615_0052/graalvm-ce-java16-linux-amd64-dev.tar.gz
  && wget "${GRAALVM_DEV_DOWNLOADS_URL}/${GRAALVM_VERSION}-${GRAALVM_GITHUB_RELEASE_TAG}/${GRAALVM_DOWNLOAD_TARGET}" \
  && tar --extract --gzip --file=${GRAALVM_DOWNLOAD_TARGET} \
  && ln --symbolic $GRAALVM_HOME /graalvm \
  \
  # Gradle
  && echo "Downloading Gradle" \
  && wget --no-verbose --output-document=gradle.zip "https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip" \
  \
  && echo "Installing Gradle" \
  && unzip gradle.zip \
  && rm gradle.zip \
  && mv "gradle-${GRADLE_VERSION}" "${GRADLE_HOME}/" \
  && ln --symbolic "${GRADLE_HOME}/bin/gradle" /usr/bin/gradle \
  \
  && echo "Testing Gradle installation" \
  && gradle --version \
  \
  # Cleanup
  && apt-get -qq clean \
  && rm --recursive --force /var/lib/apt/lists/* /tmp/* /var/tmp/* \
  && rm --force ${GRAALVM_DOWNLOAD_TARGET}
​
RUN gu install llvm-toolchain \
  && gu install python \
  && gu install ruby \
  && gu install wasm
```

```json
// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.183.0/containers/ubuntu
{
  "name": "GraalVM Ubuntu",
  "build": {
    "dockerfile": "Dockerfile",
    "args": {}
  },
​
  // Set *default* container specific settings.json values on container create.
  "settings": {
    "netbeans.jdkhome": "/graalvm",
    "java.import.gradle.java.home": "/graalvm",
    "java.home": "/graalvm",
    "terminal.integrated.env.linux": null,
    "terminal.integrated.inheritEnv": false
  },
​
  "containerEnv": {
    "GRADLE_HOME": "/opt/gradle/gradle-7.1",
    "JAVA_HOME": "/graalvm"
  },
​
  // Set remoteEnv so that tasks (Gradle extension) pick up proper JAVA_HOME env
  "remoteEnv": {
    "PATH": "${containerEnv:PATH}",
    "JAVA_HOME": "/graalvm"
  },
​
  "extensions": [
    "oracle-labs-graalvm.graalvm",
    "asf.apache-netbeans-java",
    "ms-azuretools.vscode-docker",
    "usernamehw.errorlens",
    "richardwillis.vscode-gradle"
  ]
}
```

**Comment by MartinBalin** [comment]
Reproducible in VirtualBox Windows.

**Comment by MartinBalin** [comment]
Will be fixed in upcoming release 21.2.0 of GraalVM in polyglot.cmd launcher, specific for Windows.

**Comment by GavinRay97** [comment]
> Will be fixed in upcoming release 21.2.0 of GraalVM in polyglot.cmd launcher, specific for Windows.

Thank you!
